### PR TITLE
ci: the inference worker is checked like the rest of the codebase

### PR DESCRIPTION
**Pull request by KevinBonnoron**
> **Stacked on #56** — its base is `fix/privilege-and-worker-races`, because the test suite this job runs is introduced there. Merge #56 first and GitHub will retarget this to `main`.

CI covered Biome, tsc, gofmt, go vet, the Go tests and the image builds. The 4.3k lines of Python had **no lint, no tests, and not even an import check** — while being the most concurrent and least typed part of the project, which is to say the place where a regression is both likeliest and quietest. A `.ruff_cache/` at the repo root showed ruff was being run by hand and nothing made it binding.

## What the job does

Backends import their runtime dependencies lazily, inside the methods that use them, so the base install alone is enough to assert that every one of them still imports. The job needs no torch, no onnxruntime and no CUDA, and runs in seconds.

- `ruff check inference`
- `pytest` — 26 tests: every backend module imports, the registry resolves every registered name, the app imports, and `services/ssml.py` is covered (rate parsing, prosody segmentation, tone resolution, bracket effects, edge cases), plus the eviction regression tests from #56

Ruff had two standing violations, both genuinely dead and removed here: `original = Qwen2Encoder.forward_one_step` in the CosyVoice patch, assigned and never used, and an unused `httpx` import in Voxtral's `load_model`.

## Local parity

`task test` and `task lint` now run the same checks, through `task test:inference` and `task lint:inference`. Ruff is pinned in the `dev` extra so its version cannot drift between a developer's machine and CI, and `task setup:python` installs it.

Formatting is deliberately **not** enforced: `ruff format --check` would want to reformat 21 of 36 files, which belongs in its own change rather than buried in the one that turns the checks on.

## Verification

The suite was run in a venv built from the base dependencies only — no torch, no onnxruntime, exactly what the CI job gets. 26 passed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Tests**
  - Expanded automated coverage for inference backends and SSML processing, including rate handling, tones, pauses, segmentation, imports, and invalid backend names.
  - Added continuous integration checks for inference linting and tests.

- **Documentation**
  - Updated project documentation to describe the complete test and lint commands, including inference checks.

- **Chores**
  - Improved development setup and task automation for running inference validation consistently.
  - Removed unused internal code without changing runtime behavior.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,6 +57,27 @@ jobs:
...
           CGO_ENABLED=1 go build -tags gtk3 -o /tmp/sirene-desktop ./desktop
 
+  # The worker is the most concurrent and least typed part of the codebase and had no
+  # check of any kind. Backends import their runtime dependencies lazily, inside the
+  # methods that use them, so the base install is enough to catch one that stopped
+  # importing — no torch, no CUDA, a few seconds.
+  inference:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11d5960a326750d5838078e36cf38b85af677262 # v4
```
[the remark this answers is not in the document]
Comment (KevinBonnoron) [reply .github/workflows/ci.yml:74]: Valid. `actions/checkout` leaves the workflow token in `.git/config`, and the job this PR adds runs repository code through `pytest` — including a `pip install -e` that executes the project's own build backend.

Applied to **all six** checkout steps in the workflow rather than only the flagged one. The `client` job runs package install scripts and a bundler, `go` runs `go test`, `docker` builds from the tree; the exposure is the same everywhere, and a workflow where one step is hardened and five are not invites the question of why. Verified that nothing in `ci.yml` pushes, commits or otherwise needs credentials — the only job that does is `update-flake` in `release.yml`, which is out of this PR's scope and genuinely needs them.

One comment above `jobs:` rather than six copies of the same note:

```yaml
# checkout leaves the workflow token in .git/config by default, and every job here runs
# repository code — package install scripts, go test, pytest. Nothing in this workflow
# pushes, so none of them need it.
```

On the severity, to be accurate rather than generous to myself: the repository is public and this workflow grants `contents: read`, so the token confers roughly what an anonymous clone already has. The fix is free and correct, and I would not want it read as having closed a hole that was meaningfully open.
